source/include/services/cs_CrownstoneService.h

```diff
@@ -98,6 +98,9 @@ class CrownstoneService: public Service, EventListener {
 
 
 private:
+	uint8_t _keySessionDataBuffer[sizeof(session_data_t)];
+
 	Characteristic<buffer_ptr_t>* _sessionDataCharacteristic = NULL;
+	Characteristic<buffer_ptr_t>* _sessionDataUnencryptedCharacteristic = NULL;
```

Comment:
pointer to buffer pointer? Usually double pointer indirection is smelly. Is there a reason for it here?

Reply:
It's a pointer to a characteristic, which happens to have a buffer as data type.